src/mappings/DAOTracker/mapping.ts

```diff
@@ -0,0 +1,127 @@
+import { Address, ipfs, store } from '@graphprotocol/graph-ts';
+import {
+  BlacklistDAO,
+  DAOTracker,
+  ResetDAO,
+  TrackDAO,
+} from '../../types/DAOTracker/DAOTracker';
+import {
+  BlacklistedDAO,
+  DAOTrackerContract,
+  ResetDAO as ResetDAOEntity,
+} from '../../types/schema';
+import {
+  Avatar_0_0_1_rc_31,
+  Controller_0_0_1_rc_31,
+  DAOToken_0_0_1_rc_31,
+  Reputation_0_0_1_rc_31,
+} from '../../types/templates';
+import { equalStrings } from '../../utils';
+
+export function getDAOTrackerContract(address: Address): DAOTrackerContract {
+  let daoTracker = DAOTrackerContract.load(address.toHex()) as DAOTrackerContract;
+  if (daoTracker == null) {
+    daoTracker = new DAOTrackerContract(address.toHex());
+    daoTracker.address = address;
+    let daoTrackerSC = DAOTracker.bind(address);
+    daoTracker.owner = daoTrackerSC.owner();
+    store.set('DAOTrackerContract', daoTracker.id, daoTracker);
+  }
+  return daoTracker;
+}
+
+export function handleTrackDAO(event: TrackDAO): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOTrackerContract(event.address);
+
+  let _avatar = event.params._avatar;
+  let _controller = event.params._controller;
+  let _reputation = event.params._reputation;
+  let _daoToken = event.params._daoToken;
+
+  /* TODO: uncomment when this issue is resolved https://github.com/graphprotocol/graph-node/issues/1333
+  // If the avatar hasn't been blacklisted
+  let daoTrackerSC = DAOTracker.bind(event.address);
+  if (daoTrackerSC.blacklisted(_avatar), 'latest') {
+    return;
+  }
+  */
+
+  // If the avatar already exists, early out
+  if (store.get('AvatarContract', _avatar.toHex()) != null) {
+    return;
+  }
+
+  // Tell the subgraph to start indexing events from the:
+  // Avatar, Controller, DAOToken, and Reputation contracts
+  Avatar_0_0_1_rc_31.create(_avatar);
+  Reputation_0_0_1_rc_31.create(_reputation);
+  DAOToken_0_0_1_rc_31.create(_daoToken);
+
+  // Track the Controller if it isn't a UController we're already tracking
+  if (store.get('UControllerOrganization', _controller.toHex()) == null) {
+    Controller_0_0_1_rc_31.create(_controller);
+  }
+
+  // Note, no additional work is needed here because...
```

Comment:
how about tracking the dao schemes ( e.g GenericScheme) ?

Reply:
We have no way of determining what the type of the scheme is when it's added to the controller.

Reply:
can be read from the contractInfo

Reply:
Yes but only for schemes we know the address of on startup.

Tracking non-universal schemes the subgraph already knows the address of at startup is handled.